wire-ios-data-model/Source/MLS/Migration/ProteusToMLSMigrationCoordinator.swift

```diff
@@ -148,6 +147,45 @@ public class ProteusToMLSMigrationCoordinator: ProteusToMLSMigrationCoordinating
         return .canStart
     }
 
+    /// This method is responsible for migrating all `mixed` group conversations to `mls`.
+    /// It evaluates each conversation to determine if the migration needs to be finalized (i.e: updating the protocol from `mixed` to `mls`)
+    /// or if it should first join the corresponding MLS group.
+    func migrateOrJoinGroupConversations() async {
+
+        let mlsGroupIds = await context.perform { [self] in
+            do {
+                let conversations = try ZMConversation.fetchAllTeamGroupConversations(
+                    messageProtocol: .mixed,
+                    in: context
+                )
+                return conversations.compactMap { $0.mlsGroupID }
+            } catch {
+                logger.warn("Can't fetch conversations with `mixed` protocol")
+                return [MLSGroupID]()
+            }
+        }
+
+        let mlsService = await context.perform { self.context.mlsService }
+
+        guard let mlsService else {
+            return logger.warn("can't migrate conversations to mls: missing `mlsService`")
+        }
+
+        for groupID in mlsGroupIds {
+            do {
+                if mlsService.conversationExists(groupID: groupID) {
+                    // if conversation exists we finalize migration
```

Comment:
it's TODO no?
```suggestion
                    // TODO: if conversation exists we finalize migration
```

Reply:
Yep I'm going to implement on a separate PR.